### PR TITLE
Machine Upgrade Update

### PR DESCRIPTION
**Pull request by TheDZD**
- Partial port of tgstation/-tg-station#9813

List of things not ported:
- Gibber upgrades (Gibber can now be constructed however)
- Seed extractor upgrades (Can also be constructed)
- Crusher/Recycler upgrades (See above)
- Microwave matter bin upgrade (not useful due to different cooking code)
- Ore redemption machine upgrades (well, not 100% true, we've had these since the /tg/ mining port.)
- Hydroponics trays upgrade made in the aforementioned PR.

List of things that are ported:
- Bluespace RPED - An RPED that works from a range, does not require machines to be opened, and can hold more than a regular RPED.
- Emitter upgrades - Upgrade micro-laser for a shorter firing delay, and/or the capacitor for lower power consumption.
- Monkey recycler - Upgrade matter bin to receive more monkey cubes, upgrade manipulator to decrease the number of monkeys required in order to make cubes.
- Holopad - Upgrade capacitor to increase the range that the AI's holographic projection can travel.
- Food/slime processor - It can now process multiple things at a time, and adds up the processing time of everything it is processing. Upgrade the matter bin to increase the output, upgrade the manipulator to increase processing speed.
- Ore redemption machine (might as well describe this) - Upgrade the matter bin to increase the number of sheets output per ore, upgrade the laser to increase number of points per ore, upgrade the manipulator to make it pick ores up off of its input tile faster.
- Biogenerators - Can hold 40 items by default, upgrade the matter bin to make it be able to hold more.
- Smartfridge - Can now hold up to 1500 items by default instead of 999, upgrade matter bin to make it hold more.
- Tier 4 upgrade parts - These are fairly expensive and require materials from mining, but offer the best possible upgrades for machines.

Needs testing, DNM.

### CODE REVIEW

File: code/game/machinery/kitchen/processor.dm [hunk 1]
```diff
@@ -164,28 +191,28 @@
...
+		"<span class='notice'>You turn on [src].</span>", \
+		"<span class='italics'>You hear a food processor.</span>")
+	playsound(src.loc, 'sound/machines/blender.ogg', 50, 1)
+	use_power(500)
+	var/total_time = 0
 	for(var/O in src.contents)
 		var/datum/food_processor_process/P = select_recipe(O)
 		if (!P)
-			log_admin("DEBUG: WARNING: The [O] in processor([src]) does not have a suitable recipe, but it was somehow put inside of the processor anyways.") //-rastaf0
+			log_admin("DEBUG: The [O] in processor([src]) does not have a suitable recipe, but it was somehow put inside of the processor anyways.")
```
Comment (tigercat2000) [inline code/game/machinery/kitchen/processor.dm]: uh... this should just use `log_debug("The [O]...`

File: code/game/objects/items/weapons/stock_parts.dm [hunk 2]
```diff
@@ -16,10 +16,40 @@
...
+	max_w_class = 3
+	max_combined_w_class = 800
+	works_from_distance = 1
+	primary_sound = 'sound/items/PSHOOM.ogg'
+	alt_sound = 'sound/items/PSHOOM_2.ogg'
 
 /obj/item/weapon/storage/part_replacer/proc/play_rped_sound()
 	//Plays the sound for RPED exchanging or installing parts.
-	playsound(src, 'sound/items/rped.ogg', 40, 1)
+	if(alt_sound && prob(3))
```
Comment (TheDZD) [inline code/game/objects/items/weapons/stock_parts.dm]: PSHOOM_2.ogg is great, 3x what it is on /tg/, but still reasonably low.

File: code/game/objects/items/weapons/stock_parts.dm [hunk 3]
```diff
@@ -16,10 +16,40 @@
 	display_contents_with_number = 1
 	max_w_class = 3
 	max_combined_w_class = 100
+	var/works_from_distance = 0
+	var/primary_sound = 'sound/items/rped.ogg'
```
Comment (TheDZD) [inline code/game/objects/items/weapons/stock_parts.dm]: Originally `var/pshoom_or_beepboopblorpzingshadashwoosh = 'sound/items/rped.ogg'`

**Comment by Fox-McCloud** [comment]
also worth checking out: https://github.com/tgstation/-tg-station/pull/9891

File: code/modules/reagents/Chemistry-Machinery.dm [hunk 4]
```diff
@@ -293,7 +293,8 @@
 	dispensable_reagents = list()
 	var/list/special_reagents = list(list("hydrogen", "oxygen", "silicon", "phosphorus", "sulfur", "carbon", "nitrogen", "tungsten", "water"),
 						 		list("lithium", "sugar", "copper", "mercury", "sodium","iodine","bromine"),
-								list("ethanol", "chlorine", "potassium", "aluminum","plasma", "radium", "fluorine", "iron", "silver"))
+								list("ethanol", "chlorine", "potassium", "aluminum","plasma", "radium", "fluorine", "iron", "silver"),
+								list("oil", "ash", "acetone", "saltpetre", "ammonia", "diethylamine"))
```
Comment (Fox-McCloud) [inline code/modules/reagents/Chemistry-Machinery.dm]: I'd add "fuel" to this last list.

**Comment by Fox-McCloud** [comment]
Question; I noticed you added the ability to make smartfridges, but there's no way to produce the circuit board?

Also, I realize you didn't want the ability for the gibber to be upgraded (or the recycler), but what are your thoughts on at least making them constructable? (same with the seed extractor).

**Comment by TheDZD** [comment]
@Fox-McCloud Made the gibber, recycler, and seed extractor constructable, but not upgradable. Smart fridges can now be upgraded, all of their circuit boards are available via the circuit imprinter now as well.

File: code/modules/research/designs/machine_designs.dm [hunk 5]
```diff
@@ -242,6 +242,76 @@
...
+	req_tech = list("programming" = 1)
+	build_type = IMPRINTER
+	materials = list("$glass" = 1000, "sacid" = 20)
+	build_path = /obj/item/weapon/circuitboard/smartfridge
+	category = list ("Misc. Machinery")
+
+/datum/design/monkey_recycler
+	name = "Machine Design (Monkey Recycler Board)"
+	desc = "The circuit board for a monkey recycler."
+	id = "smartfridge"
```
Comment (FalseIncarnate) [inline code/modules/research/designs/machine_designs.dm]: This seems wrong.

Also, might want to consider making the smartfridge boards work like the vendor boards (screwdriver to change mode) so you can build the different variants.

**Pull request merged**